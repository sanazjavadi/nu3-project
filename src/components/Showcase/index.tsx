import React, { FunctionComponent } from 'react';
import './index.css';

export const Showcase: FunctionComponent = ({ children }): JSX.Element =>  
<>
  <div className="App-Showcase">
    {children || <p>Nothing to show here...</p> }
  </div>
</>

export default Showcase;
