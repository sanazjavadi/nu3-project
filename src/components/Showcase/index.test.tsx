import React from 'react';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Showcase from './';

test('Showcase: Renders without children', () => {
  render(<Showcase />);
  const paragraphElement = screen.getByText(/nothing to show here.../i);
  expect(paragraphElement).toBeInTheDocument();
});

test('Showcase: Renders with children', () => {
  render(<Showcase><p>foo!</p></Showcase>);
  const paragraphElement = screen.getByText(/foo/i);
  expect(paragraphElement).toBeInTheDocument();
});

test('Showcase: Snapshot test without children', () => {
  const component = renderer.create(<Showcase />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Showcase: Snapshot test with children', () => {
  const component = renderer.create(<Showcase><p>foo!</p></Showcase>);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
