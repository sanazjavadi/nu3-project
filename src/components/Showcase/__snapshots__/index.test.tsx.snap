// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Showcase: Snapshot test with children 1`] = `
<div
  className="App-Showcase"
>
  <p>
    foo!
  </p>
</div>
`;

exports[`Showcase: Snapshot test without children 1`] = `
<div
  className="App-Showcase"
>
  <p>
    Nothing to show here...
  </p>
</div>
`;
